classes/form/enrolkey_signup_form.php

```diff
@@ -58,7 +58,7 @@ public function definition() {
         $mform->insertElementBefore($element, 'email');
 
         if ($this->signup_token_required()) {
-            $mform->addRule('signup_token', get_string('signup_missing', 'auth_enrolkey'), 'required', null, 'server');
+            $mform->addRule('signup_token', get_string('signup_missing', 'auth_enrolkey'), 'required', null, 'client');
```

Comment:
Why was this moved clientside?

Reply:
This enables the element to display a frontend message when the field is empty, it aligns with the rest of the elements on the sign-up page.
Validation is still performed in the validation() function.

Reply:
An example from core,

https://github.com/moodle/moodle/blob/38abfb6a01888145878e7175e0431a156f6f1078/login/signup_form.php#L51